src/mlpack/core/data/imputation_methods/listwise_deletion.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file listwise_deletion.hpp
+ * @author Keon Kim
+ *
+ * Definition and Implementation of the empty ListwiseDeletion class.
+ */
+#ifndef MLPACK_CORE_DATA_IMPUTE_STRATEGIES_LISTWISE_DELETION_HPP
+#define MLPACK_CORE_DATA_IMPUTE_STRATEGIES_LISTWISE_DELETION_HPP
+
+#include <mlpack/core.hpp>
+#include <cmath>
+
+using namespace std;
+
+namespace mlpack {
+namespace data {
+
+/**
+ * complete-case analysis.
+ * Removes all data for a case that has one or more missing values.
+ */
+template <typename T>
+class ListwiseDeletion
+{
+ public:
+  void Impute(const arma::Mat<T>& input,
+              arma::Mat<T>& output,
+              const T& mappedValue,
+              const size_t dimension,
+              const bool transpose = true)
+  {
+    // initiate output
+    output = input;
+    size_t count = 0;
+
+    if (transpose)
+    {
+      for (size_t i = 0; i < input.n_cols; ++i)
+      {
+         if (input(dimension, i) == mappedValue ||
+             std::isnan(input(dimension, i)))
+         {
+           output.shed_col(i - count);
```

Comment:
I think that here is might be faster to collect the list of columns that needs to be dropped, and then use non-contiguous matrix views to extract what you need to keep.  Something like this:

```
std::vector<arma::uword> colsToKeep;
colsToKeep.push_back(0);
colsToKeep.push_back(2);
colsToKeep.push_back(3);

// Only keep columns 0, 2, and 3.
arma::mat output = input.cols(colsToKeep);
```

But before you commit that change, it is probably worth running some quick tests with `mlpack_preprocess_impute` to ensure that speeds things up.  I think you will see a lot of speedup with large matrices.